File: jena/src/main/scala/eu/ostrzyciel/jelly/convert/jena/JenaConverterFactory.scala
package eu.ostrzyciel.jelly.convert.jena

import eu.ostrzyciel.jelly.core.{ConverterFactory, ProtoEncoder}
import org.apache.jena.datatypes.RDFDatatype
import org.apache.jena.graph.{Node, Triple}
import org.apache.jena.sparql.core.Quad

object JenaConverterFactory
  extends ConverterFactory[JenaEncoderConverter, JenaDecoderConverter, Node, RDFDatatype, Triple, Quad]:

  /**
   * @inheritdoc
   */
  override lazy val encoderConverter: JenaEncoderConverter = new JenaEncoderConverter()

  /**
   * @inheritdoc
   */
  override lazy val decoderConverter: JenaDecoderConverter = new JenaDecoderConverter()

File: jena/src/main/java/eu/neverblink/jelly/convert/jena/JenaConverterFactory.java
package eu.neverblink.jelly.convert.jena;

import eu.neverblink.jelly.core.JellyConverterFactory;
import org.apache.jena.datatypes.RDFDatatype;
import org.apache.jena.graph.Node;

public final class JenaConverterFactory
    extends JellyConverterFactory<Node, RDFDatatype, JenaEncoderConverter, JenaDecoderConverter> {

    @Override
    public JenaEncoderConverter encoderConverter() {
        return new JenaEncoderConverter();
    }

    @Override
    public JenaDecoderConverter decoderConverter() {
        return new JenaDecoderConverter();
    }
}
